extensions/smallrye-reactive-messaging-rabbitmq/deployment/src/main/java/io/quarkus/smallrye/reactivemessaging/rabbitmq/deployment/RabbitMQProcessor.java

```diff
@@ -0,0 +1,61 @@
+package io.quarkus.smallrye.reactivemessaging.rabbitmq.deployment;
+
+import javax.enterprise.context.ApplicationScoped;
+
+import com.rabbitmq.client.impl.CredentialsProvider;
+
+import io.quarkus.arc.deployment.AdditionalBeanBuildItem;
+import io.quarkus.arc.deployment.SyntheticBeanBuildItem;
+import io.quarkus.arc.deployment.SyntheticBeanBuildItem.ExtendedBeanConfigurator;
+import io.quarkus.deployment.Feature;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.annotations.ExecutionTime;
+import io.quarkus.deployment.annotations.Record;
+import io.quarkus.deployment.builditem.FeatureBuildItem;
+import io.quarkus.deployment.builditem.RunTimeConfigurationDefaultBuildItem;
+import io.quarkus.runtime.RuntimeValue;
+import io.quarkus.smallrye.reactivemessaging.rabbitmq.runtime.CredentialsProviderLink;
+import io.quarkus.smallrye.reactivemessaging.rabbitmq.runtime.RabbitMQRecorder;
+import io.smallrye.common.annotation.Identifier;
+
+public class RabbitMQProcessor {
```

Comment:
Is there a reason why you renamed the Processor? We try to follow some conventions for naming - even if I agree this one is a bit long, I would prefer if we keep it.

Also, I would keep the feature a separate build step and make this one specific about credentials instead of `build`

Reply:
Hahah well I started by adding `RabbitMQProcoessor` then realized `SmallRyeReactiveMessagingRabbitMQProcessor` only existed to contribute the feature build item, so I merged them.

Are you saying I should restore `SmallRyeReactiveMessagingRabbitMQProcessor` to its previous state and rename `RabbitMQProcessor` to something like `SmallRyeReactiveMessagingRabbitMQProcessorDynCreds`?

Reply:
Nope.

Keep `SmallRyeReactiveMessagingRabbitMQProcessor` and move the credentials stuff in a separate `@BuildStep` there, then drop `RabbitMQProcessor`. You can have as many build steps you want. Keeping the feature one and the credentials one separated in different methods makes for a better separation of things.

Reply:
Got it. Done!